lib/reek/source/source_code.rb

```diff
@@ -38,9 +41,9 @@ def initialize(code, description, parser = Parser::Ruby22)
       def self.from(source)
         case source
         when File     then new(source.read, source.path)
-        when IO       then new(source.readlines.join, 'STDIN')
+        when IO       then new(source.readlines.join, IO_IDENTIFIER)
         when Pathname then new(source.read, source.to_s)
-        when String   then new(source, 'string')
+        when String   then new(source, STRING_IDENTIFIER)
         end
```

Comment:
How about createing a slightly different initializer for SourceCode which makes the object itself more aware of the origin of its source? This would make it much easier to have different code paths for each case later on as well, instead of relying on the string contents of the description (This is of course definitely something for a separate PR).